spec/support/file_upload_helpers.rb

```diff
@@ -18,4 +18,14 @@ def uploaded_file(file_name)
       ActionDispatch::Http::UploadedFile.new({ tempfile: file, type: type })
     end
   end
+
+  module System
+    def submit_file(name)
+      sign_in Fabricate(:user)
```

Comment:
Why do we need to `sign_in` user inside this method 🤔

Reply:
Is there a way to start a test while being signed in already? 
This method is not a UI method, it's a helper that will call Devise building `login_as`:

```ruby
    def sign_in(user)
      login_as user, scope: :user

      user
    end
```

It would indeed simplify things if I can have a default behavior to sign in :) 
Let me know @ankitkalia1195  what you think :)

Reply:
@malparty Its okay to have login statement inside the test, or you can extract that behaviour outside like this. But in rspec we dont focus on reducing lines of code too much, test should be as readable as possible. 

```
  RSpec.configure do |config|
    config.before(:all, authenticated_user: true) do
       sign_in 
    end
  end
```

```
    it "submits the file", authenticated_user: true do
   end
```

Reply:
thanks, very great to have that actually, it helps to focus on what matters in the tests. 
I have updated the code accordingly. The sign_in method is still used directly in cases where I need to test the "user" object. 
Should be all good now :)